src/Apps/Order/Components/TwoColumnLayout.tsx

```diff
@@ -1,7 +1,37 @@
 import React from "react"
+import { Box } from "Styleguide/Elements/Box"
+import { Flex } from "Styleguide/Elements/Flex"
 import { Col, Row } from "Styleguide/Elements/Grid"
+import { Spacer } from "Styleguide/Elements/Spacer"
 import { Responsive } from "Utils/Responsive"
 
+const CONTENT_SPAN = 7
+const SIDEBAR_SPAN = 4
+const ROW_SPACE = 2
+const COL_SPACE = 2
+
+const columnRatioWidth = (spans, size) => Math.round((spans / size) * 100) + "%"
+
+export const TwoColumnSplit = ({ children, ...props }) => {
+  let [firstColumn, secondColumn] = React.Children.toArray(children)
+
+  return (
+    <Responsive>
+      {({ xs }) => (
+        <Flex flexDirection={xs ? "column" : "row"} {...props}>
+          <Box width={xs ? "100%" : columnRatioWidth(4, CONTENT_SPAN)}>
+            {firstColumn}
+          </Box>
+          <Spacer mr={xs ? null : COL_SPACE} mb={xs ? ROW_SPACE : null} />
+          <Box width={xs ? "100%" : columnRatioWidth(3, CONTENT_SPAN)}>
+            {secondColumn}
+          </Box>
+        </Flex>
```

Comment:
I almost wonder if, instead of the ternary checks for `xs`, we just had an `if`/`else` statement? It might clarify this and make it easier to read. Something like:

```jsx
if (xs) {
  return (
    <Flex flexDirection={"column"} {...props}>
      <Box width={"100%"}>{firstColumn}</Box>
      ...
    </Flex>
  )
} else {
  return (
    <Flex flexDirection={"row"} {...props}>
      <Box width={columnRationWidth(4, CONTENT_SPAN)}>{firstColumn}</Box>
      ...
    </Flex>
  )
}
```

Not sure what the conventions here are.

Reply:
Yeah, that's a good point. I'm switching on enough logic here that it probably makes the most sense to split it out.

Reply:
I'm normally a fan of turning lots of inline ternaries into one big ternary/if statement like this. Although an interesting approach for this responsive component in particular might be an api like this:

```jsx
<Responsive
  xs={{ columnWidth: "100%", flexDirection: "column" ... }}
  md={{ columnWidth: columnRatioWidth(4, CONTENT_SPAN), flexDirection: "row" ...}}
/>
  {({ columnWidth, flexDirection ... }) => (
    <Flex flexDirection={flexDirection} {...props}>
      <Box width={columnWidth}>{firstColumn}</Box>
      ...
    </Flex>
  )}
</Responsive>
```

Reply:
I'm going to tackle this in a separate PR. It's brilliant though. I love it, heh.